pkg/upgrade/control_plane_test.go

```diff
@@ -0,0 +1,71 @@
+package upgrade_test
+
+import (
+	"testing"
+
+	"github.com/go-logr/logr"
+	"github.com/vmware/cluster-api-upgrade-tool/pkg/upgrade"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type log struct {}
```

Comment:
This is one reason I don't like using logr as our interface for logging, we have to implement the whole interface and I don't believe we use the entire thing